codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/CodegenVisitor.kt

```diff
@@ -0,0 +1,282 @@
+                    .letIf(settings.codegenConfig.addMessageToErrors, AddErrorMessage::transform)
+                    .let(OperationNormalizer::transform)
+                    .let { RemoveEventStreamOperations.transform(it, settings) }
+    // .let(EventStreamNormalizer::transform)
+
+    fun execute() {
+        val service = settings.getService(model)
+        logger.info(
+                "[rust-server-codegen] Generating Rust server for service ${service}, protocol ${protocolConfig.protocol}..."
+        )
+        val serviceShapes = Walker(model).walkShapes(service)
+        serviceShapes.forEach { it.accept(this) }
+        codegenDecorator.extras(protocolConfig, rustCrate)
+        val module = RustMetadata(public = true)
+        rustCrate.withModule(RustModule("error", module)) { writer -> renderSerdeError(writer) }
+        rustCrate.finalize(settings, codegenDecorator.libRsCustomizations(protocolConfig, listOf()))
+        try {
+            "cargo fmt".runCommand(
+                    fileManifest.baseDir,
+                    timeout = settings.codegenConfig.formatTimeoutSeconds.toLong()
+            )
+        } catch (err: CommandFailed) {
+            logger.warning(
+                    "[rust-server-codegen] Failed to run cargo fmt: [${service.id}]\n${err.output}"
+            )
+        }
+
+        logger.info("[rust-server-codegen] Rust server generation complete!")
+    }
+
+    override fun getDefault(shape: Shape?) {}
+
+    override fun operationShape(shape: OperationShape?) {
+        logger.info("[rust-server-codegen] Generating operation ${shape}...")
+        if (shape != null) {
+            val inputHttpDocumentMembers =
+                httpBindingResolver.requestMembers(shape, HttpLocation.DOCUMENT)
+            val outputHttpDocumentMembers = 
+                httpBindingResolver.responseMembers(shape, HttpLocation.DOCUMENT)
+            rustCrate.useShapeWriter(shape) { writer ->
+                shape.let {
+                    httpDeserializerGenerator.render(writer, it)
+                    httpSerializerGenerator.render(writer, it)
+                    serializerGenerator.renderStructure(writer, shape.inputShape(model), inputHttpDocumentMembers)
+                    serializerGenerator.renderStructure(writer, shape.outputShape(model), outputHttpDocumentMembers)
+                    deserializerGenerator.renderStructure(writer, shape.inputShape(model), inputHttpDocumentMembers)
+                    deserializerGenerator.renderStructure(writer, shape.outputShape(model), outputHttpDocumentMembers)
+                    shape.errors.forEach { error ->
+                        val errorShape = model.expectShape(error, StructureShape::class.java)
+                        serializerGenerator.renderStructure(writer, errorShape, errorShape.members().toList())
+                        deserializerGenerator.renderStructure(writer, errorShape, errorShape.members().toList())
+                    }
```

Comment:
@rcoh I have moved the whole logic inside the CodegenVisitor and removed the inherited Json serde modules so that the client codegen ones are now back to be final and without changed visibility for their members.